lib/signs/utilities/messages.ex

```diff
@@ -122,6 +130,69 @@ defmodule Signs.Utilities.Messages do
     {bottom, top}
   end
 
+  defp get_last_trip_messages(
+         {top_message, bottom_message} = messages,
+         {has_top_service_ended?, has_bottom_service_ended?},
+         {top_source, bottom_source}
+       ) do
+    cond do
+      has_top_service_ended? and has_bottom_service_ended? ->
+        {%Content.Message.LastTrip.StationClosed{}, %Content.Message.LastTrip.ServiceEnded{}}
+
+      has_top_service_ended? ->
+        if get_message_length(bottom_message) <= 18 do
+          {bottom_message,
```

Comment:
This works fine for typical mezzanine modes, but it doesn't handle cases where the lines no longer correspond 1-1 with the source configs, like combined headways, combined alerts, or JFK special paging. I think we'll need to take a similar approach to the early AM logic, and unpack those cases so we can handle them more holistically.

Reply:
I think at this point in the logic, the only case where the lines don't correspond 1-1 anymore is the JFK Mezzanine case where we have a NB platform prediction alongside SB headways. We could account for that specific case and unpack them into a regular prediction struct and paging headway struct respectively. I think at that point, the rest of the logic here would work. What are your thoughts on that or do you see a bug in that logic that I might've missed?

Reply:
One case I ran into was if a mezzanine sign comes up with no predictions, it can display a two-line combined headway message, which can then get sliced in half if one direction is LTOD. I think a similar thing can happen with alerts, though I didn't check that definitively.

Reply:
Ah right the headways is one that doesn't work. I wonder if the alerts case is something we can just pass along though considering that if in any case a MZ sign is showing a two-lined alert message, I believe that means service is definitely not running at that station and saying "No service use shuttle bus" or just simply "No service" essentially means the same thing as something like "Station closed service has ended for the night". Might be something to run by a PM/Designer.

Reply:
Spoke with Dan and we decided that for two-line combined alerts, we should show LTOTD message once service has fully ended. Otherwise, just show the alert.